### DISCUSSION TITLE
How to release helm chart to another kubernetes cluster

### DISCUSSION THREAD

**Discussion by alex-lx** [fluxcd/flux2#1473] [Q&A] [answered]
The background
We're using [Aliyun Service Mesh](https://www.alibabacloud.com/en/product/servicemesh)(ASM) for  istio.  ASM deploys istio control plane on its own kubernetes cluster and we (the client) are only allowed to operate istio's CRDs like VirtualService on it, so we're not able to deploy a helm controller on it.
But we still want to use helm to release our istio's CRDs.
Use plain helm, [SQL storage backend](https://helm.sh/docs/topics/advanced/#sql-storage-backend) might be able to do this (while we have not tried this). 
Is It anyway possible with flux?

**Comment by stefanprodan** [comment, marked as answer]
Flux can manage Helm releases on remote clusters as long as they are accessible from the cluster where Flux runs. Docs here: https://fluxcd.io/docs/components/helm/helmreleases/#remote-clusters--cluster-api

**Reply by alex-lx** [reply]
Wow, I'll try this right now, Thanks~

**Reply by alex-lx** [reply]
Sorry to bother again, It says`The Helm storage is stored on the remote cluster`, however, we're not able to create secret or configmap on the remote cluster. Is it possible to store Helm storage on the local cluster or the [SQL storage backend](https://helm.sh/docs/topics/advanced/#sql-storage-backend)?

**Reply by hiddeco** [reply]
This is not possible at present, for SQL storage support, please open an issue in the [helm-controller](https://github.com/fluxcd/helm-controller) repository.

Can't guarantee it will be added in the near future, but would be useful to track interest.

**Reply by stefanprodan** [reply]
@alex-lx the only way to do this with Flux right now is by running `helm template` in CI and pushing the resulting manifests to a Git branch, then you can use a Flux Kustomization to reconcile the manifests on remote clusters: https://fluxcd.io/docs/components/kustomize/kustomization/#remote-clusters--cluster-api

**Reply by alex-lx** [reply]
@hiddeco Got it, hope it could be supported in some days, anyway, thanks for point out this~

**Reply by alex-lx** [reply]
@stefanprodan Thanks, I'll try it.
